src/Parsing/Impl/Ast/ScopeStatement.cs

```diff
@@ -18,9 +18,15 @@
 using Microsoft.Python.Parsing;
 
 namespace Microsoft.Python.Parsing.Ast {
-    public abstract class ScopeStatement : Statement, IScopeStatement {
+    public abstract class ScopeStatement : Statement, IScopeNode {
+        private Statement _body;
+
+        public ScopeStatement() {
+            _body = new EmptyStatement();
+        }
+        
         public IScopeNode Parent { get; set; }
-        public abstract Statement Body { get; }
+        public virtual Statement Body => _body;
```

Comment:
I think you could leave this as abstract and require every implementation to give a value, rather than adding state to the abstract base.

Reply:
And I think you should be fine to use null rather than an empty statement, but that's less important.

Reply:
Ok